core/modules/comment/comment.install

```diff
@@ -470,6 +470,45 @@ function comment_update_1007() {
   }
 }
 
+/**
+ * New setting for comment titles. Replacing comment_subject_field
+ * with comment_title_options.
+ */
+function comment_update_1008() {
+  foreach (config_get_names_with_prefix('node.type.') as $config_name) {
+    $config = config($config_name);
+    $comment_title = $config->get('settings.comment_subject_field');
+    if ($comment_title == COMMENT_TITLE_AUTO) {
+      // Check to see if contrib module is installed and preserve settings.
+      if (module_exists('comment_hide_subject')) {
+        $config->set('settings.comment_title_options', COMMENT_TITLE_HIDDEN);
+      } 
+      else {
+        $config->set('settings.comment_title_options', COMMENT_TITLE_AUTO);   
+      }
+    }
+    else {
+      $config->set('settings.comment_title_options', COMMENT_TITLE_CUSTOM);
+    }
+    $config->save();
+    // Remove old comment_subject_title setting
+    foreach (node_type_get_types() as $type_name => $node_type) {
+      if (isset($node_type->settings['comment_subject_field'])) {
+        unset($node_type->settings['comment_subject_field']);
+      }
+    }
+    node_type_save($node_type);
```

Comment:
This is unnecessarily looping through all node types within a loop that is already going through all node types. We're already working on the node type configuration above. Before calling `$config->save()`, the old value should be cleared with `$config->clear('settings.comment_subject_field')`.